flow360/component/simulation/unit_system.py

```diff
@@ -358,12 +367,18 @@ def get_class_object(cls, dim_type, **kwargs):
             """Get a dynamically created metaclass representing the constraint"""
 
             class _ConType(pd.BaseModel):
-                value: Annotated[float, annotated_types.Interval(**kwargs)]
+                kwargs.pop("allow_inf_nan", None)
+                value: Annotated[
+                    float,
+                    annotated_types.Interval(
+                        **{k: v for k, v in kwargs.items() if k != "allow_inf_nan"}
+                    ),
+                ]
 
-            def validate(con_cls, value, *args):
+            def validate(con_cls, value, *args, **kwargs):
                 """Additional validator for value"""
                 try:
-                    dimensioned_value = dim_type.validate(value)
+                    dimensioned_value = dim_type.validate(value, kwargs)
```

Comment:
shouldn't this be:
```suggestion
                    dimensioned_value = dim_type.validate(value, **kwargs)
```
?

Reply:
You are correct.